components/renku_data_services/notebooks/blueprints.py

```diff
@@ -388,14 +388,19 @@ async def _handler(
 
             base_server_url = self.nb_config.sessions.ingress.base_url(server_name)
             base_server_path = self.nb_config.sessions.ingress.base_path(server_name)
+            ui_path: str = (
+                base_server_path.rstrip("/") + "/" + environment.default_url.lstrip("/")
```

Comment:
nitpick: you can use quotes in f-strings in python 3.12, so this would work :) 
```suggestion
                f"{base_server_path.rstrip("/")}/{environment.default_url.lstrip("/")}"
```